Code/src/main/java/nl/utwente/group10/haskell/type/VarT.java

```diff
@@ -156,4 +156,9 @@ public final String toString() {
 
         return intersection;
     }
+
+    @Override
+    public int compareTo(final Type type) {
+        return this.equals(type) ? 0 : -1;
```

Comment:
You're not overriding `equals` as far as I can tell. Does that mean you want `Object.equals`, i.e. VarT's are only equal if they're the same object? That seems reasonable, but could be worthwhile to add as a comment.